modules/k8s/job.go

```diff
@@ -0,0 +1,93 @@
+// test if there is an error.
+func ListJobs(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) []batchv1.Job {
+	jobs, err := ListJobsE(t, options, filters)
+	require.NoError(t, err)
+	return jobs
+}
+
+// ListJobsE will look for jobs in the given namespace that match the given filters and return them.
+func ListJobsE(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) ([]batchv1.Job, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clientset.BatchV1().Jobs(options.Namespace).List(context.Background(), filters)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Items, nil
+}
+
+// GetJob returns a Kubernetes job resource in the provided namespace with the given name. This will
+// fail the test if there is an error.
+func GetJob(t testing.TestingT, options *KubectlOptions, jobName string) *batchv1.Job {
+	job, err := GetJobE(t, options, jobName)
+	require.NoError(t, err)
+	return job
+}
+
+// GetJobE returns a Kubernetes job resource in the provided namespace with the given name.
+func GetJobE(t testing.TestingT, options *KubectlOptions, jobName string) (*batchv1.Job, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+	return clientset.BatchV1().Jobs(options.Namespace).Get(context.Background(), jobName, metav1.GetOptions{})
+}
+
+// WaitUntilJobSucceed waits until job suceeded is superior to 0, retrying the check for the specified amount of times, sleeping
+// for the provided duration between each try. This will fail the test if there is an error or if the check times out.
+func WaitUntilJobSucceed(t testing.TestingT, options *KubectlOptions, jobName string, retries int, sleepBetweenRetries time.Duration) {
+	require.NoError(t, WaitUntilJobSucceedE(t, options, jobName, retries, sleepBetweenRetries))
+}
+
+// WaitUntilJobSucceedE waits until job suceeded is superior to 0, retrying the check for the specified amount of times, sleeping
+// for the provided duration between each try.
+func WaitUntilJobSucceedE(t testing.TestingT, options *KubectlOptions, jobName string, retries int, sleepBetweenRetries time.Duration) error {
+	statusMsg := fmt.Sprintf("Wait for job %s to be provisioned.", jobName)
+	message, err := retry.DoWithRetryE(
+		t,
+		statusMsg,
+		retries,
+		sleepBetweenRetries,
+		func() (string, error) {
+			job, err := GetJobE(t, options, jobName)
+			if err != nil {
+				return "", err
+			}
+			if !IsJobSucceeded(job) {
+				return "", NewJobNotSucceeded(job)
+			}
+			return "Job is now Succeeded", nil
+		},
+	)
+	if err != nil {
+		logger.Logf(t, "Timed out waiting for Job to be provisioned: %s", err)
+		return err
+	}
+	logger.Logf(t, message)
+	return nil
+}
+
+// IsJobSucceeded returns true if the Status Succeeded of the job is superior to 0
+func IsJobSucceeded(job *batchv1.Job) bool {
+	return job.Status.Succeeded > 0
```

Comment:
I think this should be `job.Status.Active == 0 && job.Status.Failed == 0`. Semantically, I don't think a job is successful unless all the Pods run to completion and are succeeded. There might be use cases for looking for the first success, but my guess is that that would be rare.

Reply:
I just modified it & change the test to match the new condition :+1: